routingpy/routers/heremaps.py

```diff
@@ -596,6 +596,9 @@ def directions(  # noqa: C901
             if self.api_key is None
             else "https://route.ls.hereapi.com/routing/7.2"
```

Comment:
this whole block can go actually, it's the same URL as the default from the init

Reply:
But if you first call `.matrix()` and then `.directions()`, the base_url still needs to change. So set it to a default in the init and then keep the updating block for each of the three methods?

Reply:
yeah you're right! then I guess we can take out the default URLs up top?

Reply:
you mean the logic to decide whether to use `_APIKEY_BASE_URL` or `_DEFAULT_BASE_URL`?

Reply:
yep, since we do that now for every single method? unless I'm mistaken there's no need anymore to define self.base_url in the init. and the constants up top defining the URLs are redundant then

Reply:
True, they are. We do have to pass something in the init though, since it's a required arg.

Reply:
if it's not critical for the client's init, simply pass a `''` or just some other garbage;)